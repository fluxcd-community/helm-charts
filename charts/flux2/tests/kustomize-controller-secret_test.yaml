suite: test kustomize-controller-secret deployment
templates:
  - kustomize-controller-secret.yaml
tests:
  - it: should be empty if kustomizecontroller.secret.create is false
    asserts:
      - hasDocuments:
          count: 0
    set:
      kustomizecontroller.secret.create: false
  - it: should have kind Secret for k8s >= 1.19
    capabilities:
      majorVersion: 1
      minorVersion: 19
    set:
      kustomizecontroller.secret.create: true
      kustomizecontroller.secret.name: test1
      kustomizecontroller.secret.data:
        testkey: testvalue
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - isAPIVersion:
          of: v1
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
    set:
      kustomizecontroller.secret.create: true
      kustomizecontroller.secret.name: test1
      kustomizecontroller.secret.data:
        testkey: testvalue
