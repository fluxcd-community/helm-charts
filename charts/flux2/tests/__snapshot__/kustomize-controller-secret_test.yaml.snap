should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      testkey: dGVzdHZhbHVl
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        app.kubernetes.io/version: 0.31.3
        helm.sh/chart: flux2-1.0.0
      name: test1
      namespace: NAMESPACE
    type: Opaque
